applications/sink/kafka-sink/src/main/resources/META-INF/dataflow-configuration-metadata.properties

```diff
@@ -1,2 +1,3 @@
 configuration-properties.classes=org.springframework.cloud.fn.consumer.kafka.KafkaPublisherProperties, \
-  org.springframework.boot.autoconfigure.kafka.KafkaProperties
+  org.springframework.boot.autoconfigure.kafka.KafkaProperties, \
+org.springframework.boot.autoconfigure.kafka.KafkaProperties$Producer
```

Comment:
Should we also expose the common properties they may use to configure connection to Kafka?
- KP.properties
- KP.ssl (can be done on KP.producer.security though)
- KP.security (can be done on KP.producer.security though)
- KP.retry

Reply:
I tried that and it is got a bit messy: I cannot restrict it only to those props which I need here. With `org.springframework.boot.autoconfigure.kafka.KafkaProperties$Security` it shows also a `Consumer`, `Admin` and `Streams` parts.
If that is still a goal, I can go ahead and add all of them: even if we have some extra props, we would be covered for all those we really need.

Reply:
I figured you probably explored this already. 

I have mixed feelings because KafkaProperties is so huge and many of the properties are unrelated to producing. Having more unrelated props documented may be more confusing for users than listing the producer-specific ones. I mean, all the other props will work - they just will not be advertised in the README. 

Based on that, I agree w/ the current changes.